src/Asset/File/Queue.php

```diff
@@ -51,6 +51,11 @@ public function __construct(Injector $injector, Packages $packages)
      */
     public function add(FileAssetInterface $asset)
     {
+        if (!$asset->getPackageName()) {
+            // Deprecated.
```

Comment:
Nit pick … I don't get many chances to make them in your direction :wink: 

Wouldn't a PHPDoc `@deprecated` be a better option here, it makes regex searching a bit easier during major version Rip & Replace sessions, skipping the need to check `git blame` & Co.